multi-node/aws/pkg/cluster/cluster_test.go

```diff
@@ -281,3 +283,102 @@ hostedZone: staging.core-os.net
 		t.Errorf("failed to catch already existing ExternalDNSName")
 	}
 }
+
+func TestStackTags(t *testing.T) {
+	testCases := []struct {
+		expectedTags []*cloudformation.Tag
+		clusterYaml  string
+	}{
+		{
+			expectedTags: []*cloudformation.Tag{},
+			clusterYaml: `
+#no stackTags set
+`,
+		},
+		{
+			expectedTags: []*cloudformation.Tag{
+				&cloudformation.Tag{
+					Key:   aws.String("KeyA"),
+					Value: aws.String("ValueA"),
+				},
+				&cloudformation.Tag{
+					Key:   aws.String("KeyB"),
+					Value: aws.String("ValueB"),
+				},
+				&cloudformation.Tag{
+					Key:   aws.String("KeyC"),
+					Value: aws.String("ValueC"),
+				},
+			},
+			clusterYaml: `
+stackTags:
+  KeyA: ValueA
+  KeyB: ValueB
+  KeyC: ValueC
+`,
+		},
+	}
+
+	for _, testCase := range testCases {
+
+		configBody := minimalConfigYaml + testCase.clusterYaml
+		clusterConfig, err := config.ClusterFromBytes([]byte(configBody))
+		if err != nil {
+			t.Errorf("could not get valid cluster config: %v", err)
+			continue
+		}
+
+		cluster := &Cluster{
+			Cluster: *clusterConfig,
+		}
+
+		cfSvc := &dummyCloudformationService{
+			ExpectedTags: testCase.expectedTags,
+		}
+
+		_, err = cluster.createStack(cfSvc, "")
+
+		if err != nil {
+			t.Errorf("error creating cluster: %v\nfor test case %+v", err, testCase)
+		}
+	}
+}
+
+type dummyCloudformationService struct {
+	ExpectedTags []*cloudformation.Tag
+}
+
+func (cfSvc *dummyCloudformationService) CreateStack(req *cloudformation.CreateStackInput) (*cloudformation.CreateStackOutput, error) {
+
+	if len(cfSvc.ExpectedTags) != len(req.Tags) {
+		return nil, fmt.Errorf(
+			"expected tag count does not match supplied tag count\nexpected=%v, supplied=%v",
+			cfSvc.ExpectedTags,
+			req.Tags,
+		)
+	}
+
+	matchCnt := 0
+	for _, eTag := range cfSvc.ExpectedTags {
```

Comment:
Not sure if it matters, but this would still pass in the case of duplicate tags. Pseudo passing test case:

```
want = []tag{
  {k:foo, v:bar},
  {k:foo, v:bar},
}
got = []tag{k:foo, v:bar}
```

Also, maybe reflect.DeepEqual() instead of manually checking struct fields?